types/validator_set.go

```diff
@@ -510,35 +488,32 @@ func verifyRemovals(deletes []*Validator, vals *ValidatorSet) error {
 // Should not fail as verification has been done before.
 func (vals *ValidatorSet) applyRemovals(deletes []*Validator) {
 
-	for _, valUpdate := range deletes {
-		address := valUpdate.Address
-		_, removed := vals.Remove(address)
-		if !removed {
-			// Should never happen
-			panic(fmt.Sprintf("failed to remove validator %X", address))
+	existing := make([]*Validator, len(vals.Validators))
+	copy(existing, vals.Validators)
```

Comment:
once again, what's the reason for additional structs? to make the op atomic?

Reply:
no good reason currently, in an earlier version I used to do some checks and wanted the vals.Validators to stay intact. I removed the copy now.